homeassistant/components/zha/entity.py

```diff
@@ -85,14 +91,19 @@ def device_info(self) -> DeviceInfo:
         ieee = zha_device_info["ieee"]
         zha_gateway = self.entity_data.device_proxy.gateway_proxy.gateway
 
-        return DeviceInfo(
+        device_info = DeviceInfo(
             connections={(CONNECTION_ZIGBEE, ieee)},
             identifiers={(DOMAIN, ieee)},
             manufacturer=zha_device_info[ATTR_MANUFACTURER],
             model=zha_device_info[ATTR_MODEL],
             name=zha_device_info[ATTR_NAME],
-            via_device=(DOMAIN, str(zha_gateway.state.node_info.ieee)),
         )
+        if ieee != str(zha_gateway.state.node_info.ieee):
```

Comment:
I believe there's also a `is_active_coordinator` property on the ZHA `Device`, which we could access here with `self.entity_data.device_proxy.is_active_coordinator` (or something like that). We may wanna use that instead.

Reply:
Good point. We were talking about making a followup PR that sets the `via_device` to `None` for the coordinator, to fix current installations. This PR would only prevent new ones from being hit by the bug. I think this can be addressed there.